### PR TITLE
Event timeline(cluster, node(s), partition, application)

### PR DESCRIPTION
**Pull request by jeffj6123**
Adding initial set of timeline visual events for #219 
cluster:
- upgrade
- upgrade domain
- seed node warning

application:
- upgrade
- upgrade domain
- application process exited

node:
- node down

partition:
- swap primary

another
allintimeline

### CODE REVIEW

File: src/Sfx/App/partials/timeline.html [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+            Move to oldest event
+        </button>
+        
+        <button type="button" class="checkbox-push timeline-button" ng-click="ctrl.moveToNewestEvent()"
+            title="Move to the newest event in the time range.">
+            Move to newest event
+        </button>
+
+        <a> Why is data missing?
+            <span class="bowtie-icon bowtie-status-help-outline" title="This timeline may present incomplete information and not fully represent the cluster for this date range. Events may also be missing due to data loss and lead to imcomplete states. E.g.:A cluster upgrade starting before or ending after this time range will not be shown."></span>
```
Comment (Christina-Kang) [inline src/Sfx/App/partials/timeline.html]: nit: E.g:A => E.g.: A (add a space there)

File: src/Sfx/App/Scripts/Models/eventstore/timelineGenerators.ts [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+            if (rollbackStartedEvent) {
+                rollbackStarted = rollbackStartedEvent.timeStamp;
+                const rollbackStartedDate = new Date(rollbackEnd);
+                const upgradeDuration = rollbackCompleteEvent.eventProperties["OverallUpgradeElapsedTimeInMs"];
+
+                const upgradeStart = new Date(rollbackStartedDate.getTime() - upgradeDuration).toISOString();
+                //roll forward
+                items.add({
+                    id: rollbackCompleteEvent.eventInstanceId + "upgrade",
+                    content: "upgrade rolling forward failed",
```
Comment (Christina-Kang) [inline src/Sfx/App/Scripts/Models/eventstore/timelineGenerators.ts]: nit: capitalize Upgrade?

File: src/Sfx/App/Scripts/Models/eventstore/timelineGenerators.ts [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+
+        public static parseUpgradeCompleted(event: FabricEventBase, items: vis.DataSet<vis.DataItem>, group: string, targetVersionProperty: string): void {
+            const rollBack = event.kind === "ClusterUpgradeRollbackCompleted";
+
+            const end = event.timeStamp;
+            const endDate = new Date(end);
+            const duration = event.eventProperties["OverallUpgradeElapsedTimeInMs"];
+
+            const start = new Date(endDate.getTime() - duration).toISOString();
+            const content = `${rollBack ? "upgrade Rolling back" : "Upgrade rolling forward"} to ${event.eventProperties[targetVersionProperty]}`;
```
Comment (Christina-Kang) [inline src/Sfx/App/Scripts/Models/eventstore/timelineGenerators.ts]: nit: capitalize Upgrade?

File: src/Sfx/App/Scripts/Directives/DoubleSliderDirective.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+            $scope.$watchGroup(["startDate", "endDate"], () => {
+                if ($scope.startDate &&  $scope.endDate) {
+                    //create the slider if one doesnt exist on this element
+                    //noUiSlider throws an error if you attempt to create on an element that already has it.
+                    if (!$scope.slider) {
+                        $scope.slider = noUiSlider.create(element[0], {
+                            format: {
+                                to : num => {
+                                    const date = new Date(num);
+                                    return date.toLocaleString("en-us");
```
[the remark this answers is not in the document]
Comment (jeffj6123) [reply src/Sfx/App/Scripts/Directives/DoubleSliderDirective.ts]: We dont do any intentional localization and mostly we just default to en-us style dates.

File: src/Sfx/App/Scripts/Directives/DoubleSliderDirective.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+                                },
+                                from: Number
+                            },
+                            tooltips: true,
+                            connect: true,
+                            range: {
+                                min: TimeUtils.AddDays(new Date(), -30).getTime(),
+                                max: new Date().getTime(),
+                            },
+                            step: 60 * 1000,
```
[the remark this answers is not in the document]
Comment (jeffj6123) [reply src/Sfx/App/Scripts/Directives/DoubleSliderDirective.ts]: Yeah, its minutes but I will put a label.

File: src/Sfx/App/Scripts/Directives/DoubleSliderDirective.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                            start: [new Date($scope.startDate).getTime(), new Date($scope.endDate).getTime()]
+                        });
+
+                        $scope.slider.on("set", (data) => {
+                            const end = new Date(data[1]);
+                            const start = new Date(data[0]);
+
+                            if ($scope.endDate.toUTCString() !== end.toUTCString()) {
```
[the remark this answers is not in the document]
Comment (jeffj6123) [reply src/Sfx/App/Scripts/Directives/DoubleSliderDirective.ts]: So this check is just to see if the date changed and checking utc strings is very reliable.

**Pull request merged**